datashape/coretypes.py

```diff
@@ -710,7 +710,9 @@ def _subshape(self, index):
                         (index, type(index).__name__))
 
     def __setstate__(self, state):
-        self.__init__(*state)
+        self._parameters = state
+        self.composite = True
+        self.name = None
```

Comment:
should we define `__getstate__` so that we can carry over the name attribute as well?

Reply:
actually, since there is no change in behavior this looks good to merge. We can add `name` to the state in the future if we need it.

Reply:
yeah, I was somewhat confused by this as well, because it seems like this isn't set. We can try to fix that later if we need to, but idk what uses these fields.